src/py/flwr/client/secure_aggregation/secaggplus_handler_test.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2020 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""The SecAgg+ protocol handler tests."""
+
+import unittest
+from itertools import product
+from typing import Any, Dict, List, cast
+
+from flwr.client import NumPyClient
+from flwr.common.typing import Value
+
+from .secaggplus_handler import (
+    STAGE_COLLECT_MASKED_INPUT,
+    STAGE_SETUP,
+    STAGE_SHARE_KEYS,
+    STAGE_UNMASKING,
+    STAGES,
+    SecAggPlusHandler,
+)
+
+
+class EmptyFlowerNumPyClient(NumPyClient, SecAggPlusHandler):
+    """Empty Numpy client."""
+
+
+class TestSecAggPlusHandler(unittest.TestCase):
+    """Test the SecAgg+ protocol handler."""
+
+    def test_invalid_handler(self) -> None:
+        """Test invalid handler."""
+        handler = SecAggPlusHandler()
+
+        with self.assertRaises(TypeError):
+            handler.handle_secure_aggregation({})
+
+    def test_stage_transition(self) -> None:
+        """Test stage transition."""
+        handler = EmptyFlowerNumPyClient()
+
+        assert STAGES == (
+            STAGE_SETUP,
+            STAGE_SHARE_KEYS,
+            STAGE_COLLECT_MASKED_INPUT,
+            STAGE_UNMASKING,
+        )
+
+        valid_transitions = set(
```

Comment:
It might be safer and more readable to simply hard-code all valid transitions here